mslib/msui/multiple_flightpath_dockwidget.py

```diff
@@ -0,0 +1,286 @@
+
+    :copyright: Copyright 2022 Jatin Jain
+    :copyright: Copyright 2022 by the MSS team, see AUTHORS.
+    :license: APACHE-2.0, see LICENSE for details.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+"""
+import logging
+import os
+from PyQt5 import QtWidgets, QtCore
+from mslib.msui.qt5 import ui_multiple_flightpath_dockwidget as ui
+from mslib.utils.qt import get_open_filenames
+from mslib.utils.config import load_settings_qsettings, save_settings_qsettings
+from fs import open_fs
+import xml
+
+
+class MultipleFlightpath(object):
+    """
+    Represent a Multiple FLightpath
+    """
+
+    def __init__(self, mapcanvas, wp):
+        self.map = mapcanvas
+        self.flightlevel = None
+        self.comments = ''
+        self.patches = []
+        self.waypoints = wp
+        self.draw()
+
+    def draw_line(self, x, y):
+        self.patches.append(self.map.plot(x, y, color='blue', linewidth='2'))
+
+    def compute_xy(self, lon, lat):
+        x, y = self.map.gcpoints_path(lon, lat)
+        return x, y
+
+    def get_lonlat(self):
+        lon = []
+        lat = []
+        for i in range(len(self.waypoints)):
+            lat.append(self.waypoints[i][0])
+            lon.append(self.waypoints[i][1])
+        return lat, lon
+
+    def draw(self):
+        lat, lon = self.get_lonlat()
+        x, y = self.compute_xy(lon, lat)
+        self.draw_line(x, y)
+        self.map.ax.figure.canvas.draw()
+
+    def remove(self):
+        for patch in self.patches:
+            for elem in patch:
+                elem.remove()
+        self.patches = []
+        self.map.ax.figure.canvas.draw()
+
+
+class MultipleFlightpathControlWidget(QtWidgets.QWidget, ui.Ui_MultipleViewWidget):
+    """
+    This class provides the interface for plotting Multiple Flighttracks
+    on the TopView canvas.
+    """
+
+    def __init__(self, parent=None, view=None, listView=None, activeFlightTrack=None):
+        super(MultipleFlightpathControlWidget, self).__init__(parent)
+        self.listView = listView
+        self.ui = parent
+        self.setupUi(self)
+        self.view = view  # canvas
+        self.flight_path = None  # flightpath object
+        self.dict_files = {}  # Dictionary of files added; key: patch, waypoints
+        self.waypoint_list = []  # List of waypoints parsed from FTML file
+        self.directory_location = None
+        self.waypoints_modelList = []
+        self.active_flight_track = activeFlightTrack
```

Comment:
How to update `active_flight_track` instance variable every time when new flighttrack gets activated?

Reply:
can that be readed from the UI ?
Or do you need it as a global var?

Reply:
I need global var

Reply:
Have also a look if you can solve it by emit signals, see

https://itecnote.com/tecnote/python-pyqt-modify-gui-from-another-thread/
https://doc.qt.io/archives/qt-4.8/threads-qobject.html (there may be an update of this)